main.go

```diff
@@ -81,7 +70,27 @@ func main() {
 		}))
 	}()
 
-	logger.Infof("server started at %s://%s:%d", cfg.Scheme, cfg.Host, cfg.Port)
+	logger.Infof("server started at %s://%s:%d", opts.Scheme, opts.Host, opts.Port)
+}
+
+func main() {
+	cfg, err := config.Load(os.Args[1:])
+	if err == flag.ErrHelp {
+		return
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	logger := logrus.New()
+	logger.SetLevel(cfg.LogLevel)
+
+	if cfg.Scheme != "both" {
+		startServer(logger, &cfg, cfg.Scheme)
+	} else {
+		startServer(logger, &cfg, "http")
+		startServer(logger, &cfg, "https")
+	}
```

Comment:
Can we setup two listeners that point to the same server instance? I feel like that would make the most sense.

Reply:
@fsouza , I can do that, although I am not sure there is a big penalty when having two servers.
Started implementing but ran into lister code inside `NewServerWithOptions` which complicates a bit the transition to 2 listeners with a single server:
https://github.com/ramir-savvy/fake-gcs-server/blob/ea0ba662c5381c1317c13151f231e361cb7e51e9/fakestorage/server.go#L163

`options.NoListener` is set to `true` by default and only set to `false` in the tests.
Can you clarify its use?
If the `NewServerWithOptions` function just returned in that line instead of starting listeners for tests it would be easier.
Or maybe keep it as it is and ignore that in the server run because it is relevant just for tests?

Reply:
> although I am not sure there is a big penalty when having two servers

I was under the assumption that you wanted both servers to serve the same underlying buckets and objects? Without that, the memory backend will not have the same underlying objects, and I'd prefer that we don't diverge the capabilities of the backends here.

We can also decouple the backend from the server (i.e. support taking a backend instance in the server constructor), and in that case having two servers won't matter (similar to how the grpc server takes the backend instance).

Reply:
@fsouza , you are right of course, I didn't take that into account.

I have pushed a commit that changes the code to have 2 listeners and a single server.
I looked at decoupling the backend from the server but it didn't seem to be so trivial.
Let me know what you think. Thanks